volunteercore/api/volops/opportunities.py

```diff
@@ -5,14 +5,27 @@
 from volunteercore.volops.models import Partner, Opportunity
 from volunteercore.api.errors import bad_request
 from flask_whooshalchemyplus import index_one_record
-
+from whoosh import scoring
 
 # API GET endpoint returns individual opportunity from given id
 @bp.route('/api/opportunities/<int:id>', methods=['GET'])
 @login_required
 def get_opportunity_api(id):
     return jsonify(Opportunity.query.get_or_404(id).to_dict())
 
+# Returns opportunities via a weighted search.
+# Points are scored like so: tags > name > description
+# See docs for more information:
+# https://whoosh.readthedocs.io/en/latest/recipes.html#score-results-based-on-the-position-of-the-matched-term
+def get_opportunities_score(searcher, fieldname, text, matcher):
+    print(matcher.value_as("tags"))
+    print(matcher.value_as("name"))
+    print(matcher.value_as("description"))
+    tag_match_count = matcher.value_as("tags")
+    name_match_count = matcher.value_as("name")
+    description_match_count = match.value_as("description")
+    return tag_match_count * 3 + name_match_count * 2 + description_match_count
+
```

Comment:
This is from the official whoosh docs. Tried to print the results to see what kind of data I was recieving. This is where we need more work.